### ISSUE TITLE
Apple Studio Display: "Studio Display is not responding to commands in CoreDisplay Control mode"

### ISSUE THREAD

**Issue by a1031** [alin23/Lunar#445]
I'm not sure if this is a bug or a feature, so I chose "feature" since this is related to new hardware and probably not a pre-existing software flaw.

### Name of the feature

Apple Studio Display support.

### What problem does this feature solve?

Last week I started using the new [Apple Studio Display](https://www.apple.com/studio-display/) with an M1 Pro MacBook Pro and several times over the past few days I've had this dialogue come up:

<img width="532" alt="image" src="https://user-images.githubusercontent.com/5606114/159810098-aeac23ea-98ab-46b1-bcfe-038a02e5ea8c.png">

I'm not sure whether this means Lunar would need to be updated to support the Studio Display, or if there is some incompatibility or issue with the hardware causing this message to come up.

I'm using Lunar Pro 5.5.2. When I open Lunar Preferences I see that the display is using `Apple Native` Protocol:

![image](https://user-images.githubusercontent.com/5606114/159814786-6842ea11-1897-440a-88c7-2716653acda2.png)

### My Use Case for Lunar with the Studio Display

I'd like to be able to use the brightness adjustment keys on an Apple Bluetooth Keyboard to natively control the brightness of both the Studio Display and MacBook Pro display, as well as the brightness of a Dell monitor (via `DDC`), all in sync. The behavior right now seems to be: 

#### Without Lunar running:
The Bluetooth Keyboard brightness keys control the brightness of the Studio Display only. The brightness keys on the MacBook Pro control the brightness of the MacBook Pro display only.

#### With Lunar running (default settings):

The brightness keys on both the Bluetooth Keyboard and MacBook Pro control the brightness of the MacBook Pro and Dell displays only. Neither control the Studio Display.

I just wanted to establish this issue here to start a discussion about compatibility with the Studio Display. The next time this happens I will try to identify the conditions under which the "Non-responsive display" dialogue comes up, whether the display still shows "Apple Native Protocol" at that point, etc.

@alin23 I realize that you may not have a Studio Display, or any plans to get one, so I'd be happy to help with any testing/troubleshooting if needed.

**Comment by alin23** [comment]
🤨 interesting. 

I expected this to use the same API as the Pro Display XDR. 

Can you please try some CLI commands?

First install the CLI:

```sh
/Applications/Lunar.app/Contents/MacOS/Lunar install-cli
```

Then try to get/set brightness directly through the DisplayServices framework:

```sh
# get current brightness
lunar display-services GetBrightness studio

# set brightness to 20%
lunar display-services SetBrightness studio 0.2

# set brightness to 100%
lunar display-services SetBrightness studio 1.0

# get current brightness again
lunar display-services GetBrightness studio
```

**Comment by a1031** [comment]
Ok so over the past 10 days I've experienced the `Non-responsive display "Studio Display"` dialogue pop up intermittently in two different scenarios:

Note: I have my M1 Pro MBP connected to a [Satechi Thunderbolt 4 Dock](https://satechi.net/products/thunderbolt-4-dock) via Thunderbolt 4 cable (included with the dock), and the Studio Display is connected to one of the Thunderbolt ports via Thunderbolt 3 cable (included with the display).

1. When disconnecting the Thunderbolt 4 cable from my MBP (this happens most often, but not 100% of the time).
2. When using the display, it may come up every 10-15 minutes or so (this happens less frequently). 
   Note: In this scenario, when using Lunar Manual Mode, using the brightness keys on the Bluetooth Keyboard show the brightness overlay come up on both the MBP and Studio Display, but the brightness is only actually being adjusted on the MBP - the brightness does not change unless you go below 0, in which case the Sub-zero brightness works to actually dim the display.

I've installed the CLI and run those commands:
```
❯ lunar display-services GetBrightness studio

Failed
1.0

❯ lunar display-services SetBrightness studio 0.2

Setting Brightness to 0.2 for ID: 2
Success

❯ lunar display-services SetBrightness studio 1.0

Setting Brightness to 1.0 for ID: 2
Success

❯ lunar display-services GetBrightness studio

Failed
1.0
```

**Comment by alin23** [comment]
@a1031 can you try the same commands while connecting the Studio Display directly to the MacBook, removing the Satechi dock between them?

**Comment by alin23** [comment]
This seems to be working as expected for other people using Studio Display, so maybe there's something wrong with the dock you're using.

That dialog only appears when the system can't send brightness commands to the monitor, and returns more than 5 errors in a row.

**Comment by a1031** [comment]
@alin23 Sorry, I meant to follow up on this sooner. I tried using the Studio Display connected directly to the MBP (not through the Satechi dock) for a few days, and in that scenario the dialogue only ever came up when disconnecting the Thunderbolt cable from the MBP - I did not have it come up during regular usage, so that's an improvement.

Unfortunately I was having too many other problems with the Studio Display so I ended up swapping it for a different one, experiencing the issues with the new one, then returning that one as well, and just going back to using other random LCD monitors, so I couldn't continue to test this issue. I'm holding out for a Studio Display 2nd gen or an Apple silicon 27" iMac now 😀.
